pandas/core/series.py

```diff
@@ -849,9 +849,7 @@ def __getitem__(self, key):
             return self
 
         key_is_scalar = is_scalar(key)
-        if key_is_scalar:
-            key = self.index._convert_scalar_indexer(key, kind="getitem")
-        elif isinstance(key, (list, tuple)):
+        if not key_is_scalar and isinstance(key, (list, tuple)):
```

Comment:
do we need the `not key_is_scalar` and is there some overlap with the check in L872-875

Reply:
will remove the redundant part of the check; will hold off on more invasive refactor

Reply:
probably no need then if refactoring later. I guess this is more performant at the expense of clarity. This method looks like it just needs one if key_is_scalar and an else.